packages/dcos-integration-test/extra/test_networking.py

```diff
@@ -38,25 +41,122 @@ def ensure_routable(cmd, service_points):
+                     '/opt/mesosphere/active/dcos-integration-test/util/python_test_server.py 9080'
+        app['container']['docker']['portMappings'] = [{
+            'hostPort': 0,
+            'containerPort': 9080,
+            'protocol': 'tcp',
+            'name': 'test',
+            'labels': {}
+        }]
+        if vip is not None:
+            app['container']['docker']['portMappings'][0]['labels'] = {'VIP_0': vip}
+        if network == 'USER':
+            app['ipAddress'] = {'networkName': 'dcos'}
+    app['constraints'] = [['hostname', 'CLUSTER', host]]
+    return app, uuid
+
+
+def mesos_vip_app(num, network, host, vip, ucr=False):
+    app = None
+    uuid = None
+    port = backend_port_st + num
+    if ucr is False:
+        app, uuid = get_test_app()
+    else:
+        app, uuid = get_test_app_in_ucr()
+    app['id'] = '/viptest/' + app['id']
+    app['mem'] = 16
+    app['cpu'] = 0.01
+    # define a health check that works with all the network options
+    app['healthChecks'] = [{
+        'protocol': 'MESOS_HTTP',
+        'path': '/ping',
+        'gracePeriodSeconds': 5,
+        'intervalSeconds': 10,
+        'timeoutSeconds': 10,
+        'maxConsecutiveFailures': 3,
+    }]
+    assert network != 'BRIDGE'
+    if network == 'HOST':
+        app['cmd'] = '/opt/mesosphere/bin/dcos-shell python '\
+                     '/opt/mesosphere/active/dcos-integration-test/util/python_test_server.py $PORT0'
+        app['portDefinitions'] = [{
+            'protocol': 'tcp',
+            'port': 0
+        }]
+        if vip is not None:
+            app['portDefinitions'][0]['labels'] = {'VIP_0': vip}
+        app['healthChecks'][0]['portIndex'] = 0
+    if network == 'USER':
+        app['ipAddress'] = {
+            'discovery': {
+                'ports': [{
+                    'protocol': 'tcp',
+                    'name': 'test',
+                    'number': port,
+                }]
+            }
+        }
+        app['cmd'] = '/opt/mesosphere/bin/dcos-shell python '\
+                     '/opt/mesosphere/active/dcos-integration-test/util/python_test_server.py {}'.format(port)
+        app['ipAddress']['networkName'] = 'dcos'
+        if vip is not None:
+            app['ipAddress']['discovery']['ports'][0]['labels'] = {'VIP_0': vip}
+        app['healthChecks'][0]['port'] = port
+        app['portDefinitions'] = []
+    app['constraints'] = [['hostname', 'CLUSTER', host]]
+    log.info('app: {}'.format(json.dumps(app)))
+    return app, uuid
+
+
+def vip_app(num, container, network, host, vip):
+    if container == 'UCR':
+        return mesos_vip_app(num, network, host, vip, ucr=True)
+    if container == 'DOCKER':
+        return docker_vip_app(network, host, vip)
+    assert container == 'NONE', 'unkown container option {}'.format(container)
+    return mesos_vip_app(num, network, host, vip, ucr=False)
+
+
 def vip_test(dcos_api_session, r):
```

Comment:
Not from this PR, but this paradigm gets used more in this PR, so I would prefer that we fix it now as it will greatly improve readability/reviewability.

Please do not name variables something like 'r' unless there is a clear and *limited* context. E.G. this is OK:
```
for r in response_list:
    r.raise_for_status()
```
It is clear in this case that r represents a response from that list and will only be used within the context of that iterator loop. In this PR here is a great example where this is totally fine: https://github.com/dcos/dcos/pull/1224/files#diff-89643056102d7647724e7dc94be80843R219

However, in this case, r seems to be used for no logical reason. Specifically, r is of type ```VipTest``` which has no relation to the letter ```r``` I can understand.

Later on, in totally unrelated code, this happens again with ```for r in tests:``` How was r an appropriate variable name here? https://github.com/dcos/dcos/pull/1224/files#diff-89643056102d7647724e7dc94be80843R234

So, I would recommend that we call this function like so: ```def run_vip_test(dcos_api_session, vip_test: VipTest)```
and then later on in the for-loop do: ```for vip_test in tests:```